### PR TITLE
Enable all kind of Restic repositories for backup

### PR DESCRIPTION
**Pull request by pbernery**
**This PR is not mergeable as is as it brakes existing configuration. This is currently for discussions.**

This PR changes the package so that any Restic repository can be used for backups.

The changes are quite simple to make the package universal as the code was already made to use a simple Restic repository, I've just updated the Manifest so that a repository is asked to the user instead of the SFTP settings, and the install and update scripts.

These changes have some drawbacks though: this is less easy, or should I say out-of-the-box, to set up an SFTP backup.
I chose to remove the SSH key generation because I think this should something to be done manually on the server after installation. In many case, I'm currently using Dropbox as a backup destination and I also have some setup to do (configure rclone in this case).

If this is still a desire to keep SFTP backup simple, an alternative could be to propose to generate it, through a checkbox for instance, or something more complex: a dropdown in which SFTP could be chosen, or Dropbox, and then special steps could be run when installing or updating.

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "Restic",
+  "name": "Restic for all",
```
Comment (ashemsay) [inline manifest.json:2]: Why change the name?

File: manifest.json [hunk 2]
```diff
@@ -22,54 +22,17 @@
...
     "install" : [
       {
-        "name": "server",
+        "name": "repository",
         "type": "string",
         "ask": {
-          "en": "Indicate the server where you want put your backups",
-          "fr": "Indiquez le serveur où vous voulez faire vos sauvegardes"
+          "en": "Indicate the Restic repository where you want to put your backups",
+          "fr": "Indiquez le repository Restic où vous voulez faire vos sauvegardes"
```
Comment (ashemsay) [inline manifest.json:29]: le dépôt Restic

File: manifest.json [hunk 3]
```diff
@@ -22,54 +22,17 @@
...
         },
         "help":{
-          "en": "IP address or resolvable hostname of your destination server",
-          "fr": "Adresse IP ou nom résolvable de votre serveur de destination"
+          "en": "Un repository Restic peut être un serveur SFTP, une cible rclone, etc",
```
Comment (ashemsay) [inline manifest.json:32]: A Restic repository may be a SFTP server, rclone target, etc

File: manifest.json [hunk 4]
```diff
@@ -22,54 +22,17 @@
...
+          "fr": "Un repository Restic peut être un serveur SFTP, une cible rclone, etc"
```
Comment (ashemsay) [inline manifest.json:33]: Un dépôt Restic

**Comment by ashemsay** [comment]
Since the goal is to allow use of multiple repository types, it would be nice to have a few examples in the readme with the corresponding value to set for the `repository` setting.

**Comment by ashemsay** [comment]
Add ssh key generation in case of SFTP repository

File: manifest.json (hunk 1 above)
Comment (pbernery) [reply manifest.json:2]: This was a proposition for another configuration of Restic. If you wanted to keep the package as is, it means we would have 2 Restic packages on YunoHost. I thought that having two different names would help. But as it seems that you're agreeing to update the package for a widest usage, we can keep the original name.

**Comment by pbernery** [comment]
Hey. I've just seen that you made some feedbacks, sorry for the delayed answers.

**Comment by pbernery** [comment]
> Since the goal is to allow use of multiple repository types, it would be nice to have a few examples in the readme with the corresponding value to set for the `repository` setting.

Right.

> Add ssh key generation in case of SFTP repository

Alright.

I'll update the PR in the coming days.

**Comment by ForsakenHarmony** [comment]
This would be great to have

**Comment by lapineige** [comment]
What's the status of this PR ? :)

**Comment by pbernery** [comment]
I am no longer doing such backups, thus I cannot finalize this PR correctly, I will then close it.

**Pull request closed without merging**